src/core/src/op/subtract.cpp

```diff
@@ -2,89 +2,78 @@
-        NGRAPH_TYPE_CASE(evaluate_subtract, i32, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, i64, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, u8, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, u32, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, u64, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, f16, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, f32, arg0, arg1, out, broadcast_spec);
-        NGRAPH_TYPE_CASE(evaluate_subtract, bf16, arg0, arg1, out, broadcast_spec);
-    default:
-        rc = false;
-        break;
+    template <element::Type_t ET>
+    static result_type visit(const Tensor& in0,
+                             const Tensor& in1,
+                             Tensor& out,
+                             const AutoBroadcastSpec& broadcast_spec) {
+        using T = typename element_type_traits<ET>::value_type;
+        reference::subtract(in0.data<const T>(),
+                            in1.data<const T>(),
+                            out.data<T>(),
+                            in0.get_shape(),
+                            in1.get_shape(),
+                            broadcast_spec);
+        return true;
     }
-    return rc;
-}
-}  // namespace
+};
 }  // namespace subtract
 
 // ------------------------------- v1 ------------------------------------------
-op::v1::Subtract::Subtract(const Output<Node>& arg0, const Output<Node>& arg1, const AutoBroadcastSpec& auto_broadcast)
+namespace v1 {
+Subtract::Subtract(const Output<Node>& arg0, const Output<Node>& arg1, const AutoBroadcastSpec& auto_broadcast)
     : BinaryElementwiseArithmetic(arg0, arg1, auto_broadcast) {
     constructor_validate_and_infer_types();
 }
 
-shared_ptr<Node> op::v1::Subtract::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> Subtract::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v1_Subtract_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<op::v1::Subtract>(new_args.at(0), new_args.at(1), this->get_autob());
+    return std::make_shared<Subtract>(new_args.at(0), new_args.at(1), get_autob());
 }
 
-bool op::v1::Subtract::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool Subtract::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v1_Subtract_evaluate);
-    return subtract::evaluate_subtract(inputs[0], inputs[1], outputs[0], get_autob());
+    OPENVINO_ASSERT(outputs.size() == 1);
+    OPENVINO_ASSERT(inputs.size() == 2);
+
+    outputs[0].set_shape(infer_broadcast_shape(this, inputs[0].get_shape(), inputs[1].get_shape()));
+    using namespace ov::element;
+    return IfTypeOf<bf16, f16, f32, i8, i32, i64, u8, u32, u64>::apply<subtract::Evaluate>(inputs[0].get_element_type(),
```

Comment:
Out of PR scope question - why evaluate supports i8/u8, but it isn't in `has_evaluate`?

@ilyachur @ilya-lavrenov @jane-intel

Reply:
Compared to `Add` operator the `Subtract` not supports `u16`, `i16`. 
Should be added?

Reply:
I think we need to align has_evaluate and evaluate methods at least.